base_layer/p2p/src/services/comms_outbound/service.rs

```diff
@@ -0,0 +1,225 @@
+        Self { oms }
+    }
+
+    fn send_msg(
+        &self,
+        broadcast_strategy: BroadcastStrategy,
+        flags: MessageFlags,
+        body: Frame,
+    ) -> impl Future<Item = Result<(), CommsOutboundServiceError>, Error = CommsOutboundServiceError>
+    {
+        // TODO(sdbondi): Change required when oms is async
+        future::ok(
+            self.oms
+                .send_raw(broadcast_strategy, flags, body)
+                .map_err(CommsOutboundServiceError::OutboundError),
+        )
+    }
+
+    fn forward_message(
+        &self,
+        broadcast_strategy: BroadcastStrategy,
+        envelope: MessageEnvelope,
+    ) -> impl Future<Item = Result<(), CommsOutboundServiceError>, Error = CommsOutboundServiceError>
+    {
+        // TODO(sdbondi): Change required when oms is async
+        future::ok(
+            self.oms
+                .forward_message(broadcast_strategy, envelope)
+                .map_err(CommsOutboundServiceError::OutboundError),
+        )
+    }
+}
+
+impl Service<CommsOutboundRequest> for CommsOutboundService {
+    type Error = CommsOutboundServiceError;
+    type Response = Result<CommsOutboundResponse, CommsOutboundServiceError>;
+
+    existential type Future: Future<Item = Self::Response, Error = Self::Error>;
+
+    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        Ok(().into())
+    }
+
+    fn call(&mut self, req: CommsOutboundRequest) -> Self::Future {
+        match req {
+            // Send a ping synchronously for now until comms is async
+            CommsOutboundRequest::SendMsg {
+                broadcast_strategy,
+                flags,
+                body,
+            } => Either::A(self.send_msg(broadcast_strategy, flags, *body)),
+            CommsOutboundRequest::Forward {
+                broadcast_strategy,
+                message_envelope,
+            } => Either::B(self.forward_message(broadcast_strategy, *message_envelope)),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crossbeam_channel as channel;
+    use futures::Async;
+    use rand::{distributions::Alphanumeric, rngs::OsRng, Rng};
+    use std::iter;
+    use tari_comms::{
+        connection::NetAddress,
+        message::{MessageEnvelopeHeader, NodeDestination},
+        outbound_message_service::OutboundMessage,
+        peer_manager::{NodeId, NodeIdentity, Peer, PeerFlags, PeerManager},
+        types::CommsPublicKey,
+    };
+    use tari_crypto::keys::PublicKey;
+    use tari_storage::{key_val_store::lmdb_database::LMDBWrapper, lmdb_store::LMDBBuilder};
+    use tari_utilities::message_format::MessageFormat;
+    use tempdir::TempDir;
+
+    pub fn random_string(len: usize) -> String {
+        let mut rng = OsRng::new().unwrap();
+        iter::repeat(()).map(|_| rng.sample(Alphanumeric)).take(len).collect()
+    }
+
+    // TODO: Thankfully, this won't be needed in 'Future' :P - Remove this once the OMS is a Sink.
```

Comment:
...man I am not sure I can `Approve` of this...

Reply:
Puns rule!